### PR TITLE
Compat: Convert pipeline creation tests to shader module creation tests.

### PR DESCRIPTION
**Pull request by greggman**
The spec recently changed that they should be shader module creation errors.

see: https://github.com/gpuweb/gpuweb/issues/4568

Note: these tests are skipped in core. They will fail in compat until implementations update to match the changes in the spec.

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts
Comment (jrprice) [inline src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts:1]: Was this file to this PR added accidentally? I can't see anything obviously relevant to compat validation here...

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts
Comment (jrprice) [inline src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:1]: Ditto - accidental add?

File: src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/textureSample.spec.ts:1]: Yes, sorry, fixed.

File: src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts
Comment (greggman) [reply src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts:1]: yea, sorry. fixed.

**Review by jrprice — APPROVED** [review]
LGTM.

I feel like it'd be nice if there was an easy way to use non-compat as a control case for these tests to avoid typos in shaders causing failures for the wrong reasons, but there's only a handful of these tests so maybe not that important.

**Comment by greggman** [comment]
I can change all the compat tests so they are expected to pass in core. I'm make a PR

**Pull request merged**